ambry-cloud/src/main/java/com/github/ambry/cloud/azure/AzureBlobDataAccessor.java

```diff
@@ -180,14 +187,17 @@ public boolean uploadIfNotExists(BlobId blobId, long inputLength, CloudBlobMetad
    * @throws BlobStorageException for any error on ABS side.
    * @throws IOException for any error with supplied data stream.
    */
-  public void uploadFile(String containerName, String fileName, InputStream inputStream)
-      throws BlobStorageException, IOException {
+  public void uploadFile(String containerName, String fileName, InputStream inputStream) throws Exception {
     try {
-      storageClient.uploadWithResponse(containerName, fileName, true, inputStream, inputStream.available(), null, null,
-          null, null, defaultRequestConditions, uploadTimeout);
-    } catch (UncheckedIOException e) {
-      // error processing input stream
-      throw e.getCause();
+      storageClient.uploadAsyncWithResponse(containerName, fileName, true, inputStream, inputStream.available(), null,
+          null, null, null, defaultRequestConditions).get(uploadTimeout.toMillis(), TimeUnit.MILLISECONDS);
```

Comment:
Does the async client have a way to pass a timeout in as an argument? Sometimes, waiting on a future with a timeout can leave resources still in use in the background as opposed to a "deep" timeout.

Reply:
I tried checking but didn't find a way to pass timeout argument to async client. I was thinking that I might have to remove waiting on the response here in subsequent PRs since the `AzureBlobDataAccessor` APIs also would return `future`. But in that case too, I was thinking of completing the future exceptionally after a timeout which might leave resources in background. Just wondering if it is possible to clean resources when using `CompletableFutures` since they could be completed from different places.